libraries/AP_Math/quaternion.cpp

```diff
@@ -118,6 +118,271 @@ void Quaternion::from_rotation_matrix(const Matrix3f &m)
     }
 }
 
+// create a quaternion from a given rotation
+void Quaternion::from_rotation(enum Rotation rotation)
```

Comment:
Shouldn't this be a static structure?

Reply:
I think we discussed this on the dev call and seems like it's OK like this.